File: RestaurantManagementApp/src/main/java/com/example/RestaurantManagementApp/Service/RestaurantService.java
package com.example.RestaurantManagementApp.Service;

import com.example.RestaurantManagementApp.Entity.Restaurant;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface RestaurantService {

    List<Restaurant> getAllRestaurants();

    ResponseEntity<Restaurant> getRestaurantByID(int restaurantID);

    Restaurant addRestaurant(Restaurant restaurant);

    ResponseEntity<Restaurant> updateRestaurant(int id, Restaurant restaurant);

    String deleteRestaurant(int ID);

}

File: foodplatform/src/Services/RestaurantService.js
import axios from "axios";

const RESTAURANT_BASE_API_URL = "http://localhost:8080/restaurants";

class RestaurantService {
    getAllRestaurants(){
        return axios.get(RESTAURANT_BASE_API_URL);
    }

    addRestaurant(restaurant){
        return axios.post(RESTAURANT_BASE_API_URL, restaurant);
    }

    getRestaurantByID(restaurantId){
        return axios.get(RESTAURANT_BASE_API_URL +"/"+ restaurantId);
    }

    updateRestaurant(restaurantId, restaurant){
        return axios.get(RESTAURANT_BASE_API_URL +"/" + restaurantId, restaurant);
    }

    deleteRestaurant(restaurantId){
        return axios.delete(RESTAURANT_BASE_API_URL + "/" + restaurantId);
    }

}


export default new RestaurantService();